zenith_metrics/src/lib.rs

```diff
@@ -67,15 +67,21 @@ fn update_io_metrics() {
         ru_nvcsw: 0 as c_long,
         ru_nivcsw: 0 as c_long,
     };
-    unsafe {
-        getrusage(RUSAGE_SELF, (&mut usage) as *mut rusage);
+
+    let rusage_return_code = unsafe { getrusage(RUSAGE_SELF, (&mut rusage_stats) as *mut rusage) };
+    if rusage_return_code != 0 {
+        log::error!(
+            "Failed to invoke rusage, return code: {}",
+            rusage_return_code
+        );
+        return;
     }
```

Comment:
I still think assert would be a better fit here (per my previous comment).

Reply:
Sure, I'm somewhat reluctant to panics in general, but metrics code is not that important, I guess.
On the bright side, one less dependency to drag in with this approach :)